libraries/AP_BattMonitor/AP_BattMonitor_Logging.cpp

```diff
@@ -45,3 +49,106 @@ void AP_BattMonitor_Backend::Log_Write_BCL(const uint8_t instance, const uint64_
+{
+    send_smart_battery_info(instance, false, chan);
+}
+
+void AP_BattMonitor_Backend::send_smart_battery_info(const uint8_t instance, bool log_flag, const mavlink_channel_t chan)
+{
+#if HAL_SMART_BATT_LOG_ENABLED
+#ifndef HAL_NO_LOGGING
+    uint8_t cells_in_series;
+    const bool got_cells_in_series = get_cells_in_series(cells_in_series);
+
+    // get_cells_in_series only returns false if cell_count has not been determined yet
+    // if so set flag to call back logging or mavlink_send when cell_count is fixed
+    if (!got_cells_in_series) {
+        _instance = instance;
+        if (log_flag) {
+            _log_smart_batt_info = true;
+            return;
+        }
+        _mavlink_smart_batt_info = true;
+        _chan = chan;
+        return;
+    }
+
+    uint16_t cycles;
+    const bool got_cycle_count = get_cycle_count(cycles);
+
+    int32_t capacity_design;
+    const bool got_capacity_design = get_design_capacity(capacity_design);
+
+    int32_t capacity_full;
+    const bool got_capacity_full = get_full_charge_capacity(capacity_full);
+
+    int32_t design_voltage;
+    const bool got_design_voltage = get_design_voltage(design_voltage);
+
+    char serial_number[MAVLINK_MSG_SMART_BATTERY_INFO_FIELD_SERIAL_NUMBER_LEN]{};
+    get_serial_number(serial_number, ARRAY_SIZE(serial_number));
+
+    char product_name[MAVLINK_MSG_SMART_BATTERY_INFO_FIELD_DEVICE_NAME_LEN]{};
+    get_product_name(product_name, ARRAY_SIZE(product_name));
+
+    char manufacture_date[MAVLINK_MSG_SMART_BATTERY_INFO_FIELD_MANUFACTURE_DATE_LEN]{};
+    get_manufacture_date(manufacture_date,  ARRAY_SIZE(manufacture_date));
+
+    if (log_flag) {
+        struct log_BATI pkt{
+            LOG_PACKET_HEADER_INIT(LOG_BATI_MSG),
+            time_us             : AP_HAL::micros64(),
+            instance            : instance,
+            function            : MAV_BATTERY_FUNCTION_UNKNOWN,
+            type                : MAV_BATTERY_TYPE_UNKNOWN,
+            cells_in_series     : (uint8_t) (got_cells_in_series ? cells_in_series : 0),
+            capacity_design     : got_capacity_design ? capacity_design : -1,
+            capacity_full       : got_capacity_full ? capacity_full : -1,
+            design_voltage      : got_design_voltage ? design_voltage : -1,
+            cycles              : (uint16_t) (got_cycle_count ? cycles : UINT16_MAX),
+            serial_number       : {},
+            manufacture_date    : {},
+            product_name        : {},
+        };
+
+        strncpy_noterm(pkt.serial_number, serial_number, sizeof(pkt.serial_number));
+        strncpy_noterm(pkt.manufacture_date, manufacture_date, sizeof(manufacture_date));
+        strncpy_noterm(pkt.product_name, product_name, sizeof(pkt.product_name));
+        AP::logger().WriteBlock(&pkt, sizeof(pkt));
+        return;
+    }
+#endif // HAL_NO_LOGGING
+
+#ifndef HAL_NO_GCS
```

Comment:
Having a GCS with no logging won't work as all your private variables are inside the above no logging.

Reply:
Doh fixed!